packages/agreement-process/src/routers/AgreementRouter.ts

```diff
@@ -592,6 +593,36 @@ const agreementRouter = (
         return res.status(errorRes.status).send(errorRes);
       }
     })
+    .post(
+      "/internal/agreement/:agreementId/signedContract",
+      async (req, res) => {
+        const ctx = fromAppContext(req.ctx);
+
+        try {
+          validateAuthorization(ctx, [INTERNAL_ROLE]);
+
+          const { agreementId } = req.params;
+          const agreementContract = AgreementSignedContract.parse(req.body);
```

Comment:
There should be no need of a runtime type-check of the body here. Zodios already handles that at a lower API layer.